tests/with_thiserror_tests.rs

```diff
@@ -68,3 +68,31 @@ fn test_errordoc_for_struct() {
     }
     assert_display(MyError { variant: 42 }, "I'm a doc for MyError");
 }
+
+#[test]
+fn test_thiserror_implicit_and_source_works() {
+    #[derive(Display, Error, Debug)]
+    #[error("implicit source")]
```

Comment:
these are only passing because you're using `error` to derive the display which causes `thiserror` to do it.

Try doing an enum like this as a test case

```rust
#[derive(Debug, Display, Error)]
enum SourceError {
    #[error(transparent)]
    #[display(transparent)]
    ImplicitSource {
        source: io::Error,
    },
    #[error(transparent)]
    #[display(transparent)]
    ExplicitSource {
        source: String,
        #[source]
        io: io::Error,
    },
    /// There isnt really a {source}
    DocSourceless {
         source: String,
    },
}
```